ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/statetransition/epoch/AbstractEpochProcessor.java

```diff
@@ -122,7 +122,7 @@ protected void processEpoch(final BeaconState preState, final MutableBeaconState
     processRegistryUpdates(state, validatorStatuses.getStatuses());
     processSlashings(state, validatorStatuses);
     processEth1DataReset(state);
-    processPendingBalanceDeposits(state);
+    processPendingDeposits(state);
```

Comment:
I thought about the mutating `ValidatorStatuses` and seems an additional complication to me.
At current stage I'd just approach the "problem" with verbose comments.
We could be noisy here saying that here `processPendingDeposits` may add validators in the state and `ValidatorStatuses` won't contain these new added vals.